### PR TITLE
Pr add OF sensor no gyro

### PR DESCRIPTION
**Pull request by EliaTarasov**
Add support for OF sensor with no gyro onboard. Tested with `iris_opt_flow`.

### CODE REVIEW

File: src/gazebo_opticalflow_plugin.cpp [hunk 1]
```diff
@@ -117,6 +117,24 @@ this->camera = this->parentSensor->Camera();
     gzwarn << "[gazebo_optical_flow_plugin] Using default output rate " << output_rate_ << ".";
   }
 
+  bool has_gyro = HAS_GYRO;
```
Comment (TSC21) [inline src/gazebo_opticalflow_plugin.cpp]: Shouldn't this be dependent of the gyro variance and setup a threshold that would approve the gyro values according to the noise level? I think it's more reasonable than a parameter to turn off or on gyro data.

Comment (EliaTarasov) [reply src/gazebo_opticalflow_plugin.cpp]: I partially agree but on the firmware side there is strictly separation PX4/ecl#502 between NAN and other gyro values. And there is assumption that user makes this separation by providing NAN. So this proposed PR is in compliance with firmware side.

Comment (EliaTarasov) [reply src/gazebo_opticalflow_plugin.cpp]: @TSC21 So do you think we should proceed with gyro variance and threshold then?

Comment (TSC21) [reply src/gazebo_opticalflow_plugin.cpp]: I don't see where it is the user that sets the values as NAN. Can you please point me where that's stated?

Comment (EliaTarasov) [reply src/gazebo_opticalflow_plugin.cpp]: https://github.com/PX4/ecl/blob/master/EKF/optflow_fusion.cpp#L551

Comment (TSC21) [reply src/gazebo_opticalflow_plugin.cpp]: That still doesn't answer: `gyroXYZ` is gyro data, and I don't see anywhere on the code the possibility of the user setting those values to NaN.

Comment (EliaTarasov) [reply src/gazebo_opticalflow_plugin.cpp]: > gyroXYZ is gyro data, and I don't see anywhere on the code the possibility of the user setting those values to NaN.

Yes, user cannot set NAN for gyro data explicitly let's say as firmware settings. But providing gyro's data as NAN it forces firmware to rely on PX4 gyro only.

Comment (TSC21) [reply src/gazebo_opticalflow_plugin.cpp]: Yes of course, but in that case both option should be available: hard set it and have also a way of simulating bad gyro data.

Comment (EliaTarasov) [reply src/gazebo_opticalflow_plugin.cpp]: Ok i will take a look at the second option.

**Comment by mhkabir** [comment]
The whole idea of supporting NaNs in the message is to enable/test flow sensors which do not have an onboard gyroscope. There is no connection to bad gyro data.

**Comment by EliaTarasov** [comment]
Anyway i added support for onboard gyro for px4flow camera. This gyro has default gazebo [noise](http://gazebosim.org/tutorials?tut=sensor_noise) model so it can be used to model bad gyro data.

**Review by TSC21 — APPROVED** [review]

File: models/px4flow/px4flow.sdf.jinja [hunk 2]
```diff
@@ -43,7 +43,7 @@
       </collision>
 
       <sensor name="px4flow" type="imu">
-        <pose>0 0 0 0 0 0</pose>
+        <pose>0 0 0 -1.5707963259 3.141592652</pose>
```
Comment (TSC21) [inline models/px4flow/px4flow.sdf.jinja]: You are missing a dimension here

Comment (EliaTarasov) [reply models/px4flow/px4flow.sdf.jinja]: Done.

**Review by TSC21 — APPROVED** [review]
LGTM

**Pull request merged**